mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -352,9 +353,13 @@ static bool isDimOpValidSymbol(ShapedDimOpInterface dimOp, Region *region) {
 
   // Conservatively handle remaining BlockArguments as non-valid symbols.
   // E.g. scf.for iterArgs.
-  if (llvm::isa<BlockArgument>(dimOp.getShapedValue()))
-    return false;
-
+  if (auto blockArgument =
+          llvm::dyn_cast<BlockArgument>(dimOp.getShapedValue())) {
+    if (!llvm::isa<FunctionOpInterface>(
+            blockArgument.getParentRegion()->getParentOp())) {
```

Comment:
Functions may have blocks other than the entry block. Not all block arguments are function arguments, so this change looks suspicious to me.

Reply:
You're right.Thanks for the advice, I probably already know how to do it.Thanks!

Reply:
I'm a bit confused, for the block bb0, if it's parameter is a memref, then it's dimensions can change as well, but it shouldn't cause an effect like the one inside the example you gave, I'm not very sure. I think this needs to be confirmed.

Reply:
I think I've figured it out, and I'll modify the patch later.